src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -72,6 +72,18 @@ extends AbstractMap[A, B]
     else Some(e.value)
   }
 
+  override def getOrElseUpdate(key: A, defaultValue: => B): B = {
+    val i = index(elemHashCode(key))
+    val entry = findEntry0(key, i)
+    if (entry != null)
+      entry.value
+    else {
+      val newEntry = createNewEntry(key, defaultValue)
+      addEntry0(newEntry, i)
+      newEntry.value
+    }
+  }
```

Comment:
This implementation looks fine, but what about using `findOrAddEntry` in `HashTable`?  Does that still show the same speed improvement?